phpstan.neon.dist

```diff
@@ -38,6 +38,11 @@ parameters:
         # Ignore the wrong return type hint of the UrlGeneratorInterface::generate() method
         - '#Method Contao\\CoreBundle\\Picker\\AbstractPickerProvider::generateUrl\(\) never returns null so it can be removed from the return type\.#'
 
+        # Ignore wrong class-string when testing the guard that enforces it at runtime
+        -
+            message: '#Parameter \#1 \$className of method Contao\\CoreBundle\\Cache\\EntityCacheTags::getTagForModelClass\(\) expects class-string<Contao\\Model>, string given\.#'
```

Comment:
This looks fishy. PHPStan handles generics very well, so this should rather be fixed than ignored, shouldn't it?

Reply:
The problem is: PHPStan follows the annotations in a stricter way than programmers do, in fact in our setting they are treated as truth even if PHP doesn't enforce them at runtime. In this case the guard should IMHO be there for good DX, so that you can a clear exception if the passed class string isn't a model class.

Reply:
Can we make sure to only ignore the warning in the unit test and not anywhere else?

Reply:
It's already scoped to the exact file.

Reply:
Right, sorry 🙈 _Didn’t see the line after the comment ☺️_

Reply:
I think that is what the `@phpstan-ignore-next-line` annotation is for. I have already changed that in 72927064aa24f0483c1401be8eef38faa1a417c0.

Reply:
:+1: Totally forgot about that one.